tests/compile-fail/overflow_check_conditional.rs

```diff
@@ -0,0 +1,13 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(overflow_check_conditional)]
+
+fn main() {
+	let a: u32 = 1;
+	let b: u32 = 2;
+	if a + b < a {
+
+	}
```

Comment:
Rustc has several types of AST for different step of compilation. The `ast::*` stuff is just for `EarlyLintPass`s and has no type information yet, you need to use `hir::*` stuffs and a `LateLintPass` which use slightly different names for everything since a few weeks (usually when `ast` has `FooKind::Bar`, `hir` has `FooBar`). You won’t need `ast::*` stuffs here.